azure/Guidelines.md

```diff
@@ -0,0 +1,243 @@
+
+## Swagger to describe API
+
+All Services **MUST** provide as [OpenAPI Specification] that describes their service. The OpenAPI Specification is a key element of the Azure SDK plan and essential to improving the usability and discoverability of services.
+
+## URL structure
+
+In addition to the URL structure guidance in the Microsoft API guidelines, Azure has specific guidance about service exposure for multi-tenant services
+
+### URL structure
+
+All services **MUST** expose their service to developers via the following URL pattern:
+
+```
+http(s)://<service>.azure.net/<unit-of-multi-tenancy>/<service-defined-root>
+```
+
+Where:
+
+* **service** - the name of the service such as "blobstore", "servicebus", "directory", or "management"
+* **service-defined-root** - the root of the service-specific path, such as "blobcontainer", "myqueue", etc.
+* **unit-of-multi-tenancy** - refers to a globally unique moniker that identifies a unique container in the Azure service that has the following properties:
+
+  * This container is the boundary of isolation between different tenants of the service.
+  * Quotas as set and enforced at the level of this container - but there will be different limits for different operations; and operations will be service specific.
+  * Resources in the service are attached to this container and are tied to this container in terms of lifecycle. For example someone signs up, they get this container. If they unsubscribe (or don’t pay their bills) then cleanup of this container occurs and the resources associated with this container are cleaned up. Cleanup follows a state machine – the container and the resources attached to it are deactivated first (and can be easily restored if required), and if no response for some period then deleted.
+  * It is the container for billing – which means the owner of this container sees one bill for the resource usage of all azure services under this container’s identifier.
+
+For Azure PaaS services like SQL Azure, Azure Storage, Caching, etc., the unit of multi-tenancy is the Azure subscription id, which is a GUID. This ensures consistent access using the same URL pattern, and identifier, across all these services.
+
+When services produce URLs in response headers or bodies, they **MUST** use a consistent form – either always a GUID for tenant identifier or always a single verified domain - regardless of the URL used to reach the resource.
+
+### Direct endpoint URLs
+
+In addition to the required format above, services **MAY** also choose to expose direct endpoint for performance or routing reasons.  The direct endpoint should be discoverable by clients, to ensure that developers are presented with a consistent pattern for accessing Azure services.
+
+The format of the root of the direct endpoint **MUST** be as follows:
+
+```
+http(s)://<tenant-id>-<service-defined-root>.<service>.azure.net
```

Comment:
Same question throughout: https is required, and this nomenclature seems to make it optional.

Reply:
Adjusting throughout